cargo-concordium/src/main.rs

```diff
@@ -625,32 +717,225 @@ pub fn main() -> anyhow::Result<()> {
+    concordium_wasm::utils::strip(&mut rebuilt_skeleton);
+
+    let mut sha_origin = sha2::Sha256::new();
+    let mut sha_new = sha2::Sha256::new();
+    skeleton.output(&mut sha_origin)?;
+    rebuilt_skeleton.output(&mut sha_new)?;
+    let error_style = ansi_term::Color::Red.bold();
+    if sha_origin.finalize() != sha_new.finalize() {
+        anyhow::bail!(
+            "\n{}",
+            error_style.paint("The source does not correspond to the module.")
+        );
+    } else {
+        let success_style = ansi_term::Color::Green.bold();
+        eprintln!("\n{}", success_style.paint("Source and module match."));
+        Ok(())
+    }
+}
+
+fn handle_edit(options: EditOptions) -> anyhow::Result<()> {
+    let module = WasmModule::from_file(&options.source)?;
+    let mut skeleton = parse_skeleton(module.source.as_ref())
+        .context("The supplied module is not a valid Wasm module")?;
+
+    let mut build_context_section = None;
+    for (i, ucs) in skeleton.custom.iter_mut().enumerate() {
+        let cs = concordium_wasm::parse::parse_custom(ucs)?;
+        if cs.name.as_ref() == utils::BUILD_INFO_SECTION_NAME
+            && build_context_section.replace((i, cs)).is_some()
+        {
+            anyhow::bail!(
+                "Multiple sections {}. The module is malformed.",
+                utils::BUILD_INFO_SECTION_NAME
+            );
+        }
+    }
+    let Some((i, mut cs)) = build_context_section else {
+        anyhow::bail!("No embedded build information found.");
+    };
+    skeleton.custom.remove(i);
+    let mut info: utils::VersionedBuildInfo =
+        from_bytes(cs.contents).context("Failed parsing build info")?;
+    let utils::VersionedBuildInfo::V0(ref mut inner) = info;
+    if let Some(ref link) = options.source_link {
+        if options.verify {
+            eprintln!("Verifying data consistency.");
+            let mut hasher = sha2::Sha256::new();
+            download_file_into(link, &mut hasher)?;
+            anyhow::ensure!(
+                hashes::Hash::from(<[u8; 32]>::from(hasher.finalize())) == inner.archive_hash,
+                "The embedded archive hash does not match the file at the supplied URL."
+            );
+            eprintln!("The archive at the supplied URL matches the stored archive checksum.");
+        }
+        eprintln!("Setting source link to {link}.");
+    } else {
+        eprintln!("Unsetting source link.")
+    }
+    inner.source_link = options.source_link;
+    let mut out_buf = Vec::new();
+    skeleton
+        .output(&mut out_buf)
+        .context("Failed to write output module.")?;
+
+    let new_section_content = to_bytes(&info);
+    cs.contents = &new_section_content;
+    write_custom_section(&mut out_buf, &cs)?;
+
+    let out_module = WasmModule {
+        version: module.version,
+        source:  out_buf.into(),
+    };
+    std::fs::write(
+        options.source,
+        concordium_base::common::to_bytes(&out_module),
```

Comment:
I would prefer to keep it since there are two `to_bytes` methods and this makes it clear which one is used.

I know that in this case only one will work.

Reply:
Right, that makes sense. I added the comment since it is used unqualified on line 859, so I thought they should be used in the same way to make it clear that it is the same function. 
Or is this the big-endian version?
`let new_section_content = to_bytes(&info);` 
I only see the little-endian version imported.

Reply:
That one is the "little endian" one.

That's an oversight, I will change this to a qualified one.